### ISSUE TITLE
Implement guild chunker internally

### ISSUE THREAD

**Issue by davfsa** [hikari-py/hikari#247]
At the time of writing this, the `GuildChunker` is not used internally anywhere. It should be used when requesting the chunks on the initial `GUILD CREATE`. This might require a rewrite on how it works.

As an addition. `hikari.api.chunker` should be renamed to `hikari.api.guild_chunker`

**Comment by nekokatt** [comment]
I think a safe way of implementing this would be using a chunking descriptor.

Something similar to this:


```py
class GuildChunks:
    # defaults to None until the first chunk is received.
    # after that, contains the remaining chunk indicies
    # to return.
    # empty set implies completion, None implies not
    # started.
    _remaining_frames: Optional[set[int]] = None
    # map, instead, maybe?
    _future: asyncio.Future[Collection[Member]]
    
    def __await__(self):
        return self._future.__await__()
```

This object can be cached for the duration of the request, which means requesting chunks for a guild while already waiting would not send a duplicate request, but just share the awaiting GuildChunks.

This object would also contain the event listener.

**Comment by FasterSpeeding** [comment]
> This object can be cached for the duration of the request, which means requesting chunks for a guild while already waiting would not send a duplicate request, but just share the awaiting GuildChunks

What about if ppl are specifying a name pattern or array of ids for their chunk request instead of just requesting every member

**Comment by nekokatt** [comment]
@FasterSpeeding true. In this case it would be sensible to rechunk, I guess.

**Comment by FasterSpeeding** [comment]
btw what memory implications could having the lib store a member twice on start up with one being stored dependent on whether something has been marked as complete or timed-out here and what kind of approach might be suitable for marking events as timed-out on such an arbitrarily large scale

**Comment by davfsa** [comment]
Closed by #424
